docs/1.getting-started/6.data-fetching.md

```diff
@@ -264,6 +264,70 @@ const { data, error, refresh } = await useFetch('/api/users', {
 })
 ```
 
+Note that **watching a reactive element won't change the URL fetched**. For example, this will keep fetching the same initial ID of the user because the URL is constructed at the moment the function is invoked, even if parameters or a query is passed as an option.
+
+```ts
+const id = ref(1)
+
+const { data, error, refresh } = await useFetch(`/api/users/:user_id`, {
+  params: {
+    user_id: id.value
+  },
+  watch: [id]
+})
+```
+
+If you need to change the URL based on reactive elements, you may use a [reactive URL](#reactive-url) instead.
+
+#### Reactive URL
+
+Sometimes you may need to compute an URL from reactive elements, and refresh the data each time these change. Instead of juggling your way around, you can attach each param as a reactive element. Nuxt will automatically use the reactive value and re-fetch each time it changes.
+
+```ts
+const id = ref(null)
+
+const { data, pending } = useLazyRefresh('/api/user/:user_id', {
+  user_id: id
+})
+```
+
+For the case of query parameters, or complex URL construction, you may use callback to use as a [computed getter](https://vuejs.org/guide/essentials/computed.html) that returns the URL string.
+
+Every time a dependency changes, the data will be fetched using  the newly constructed URL. Combine this with [not-immediate](#not-immediate), and you can wait until the reactive element changes before fetching.
+
+```vue
+<script setup lang="ts">
+const id = ref(null)
+
+const { data, pending, status } = useLazyRefresh(() => {
+  return `/api/users?id=${id.value}`
+}, {
+  immedidate: false
+})
```

Comment:
```suggestion
const { data, pending, status } = useLazyFetch(() => `/api/users/${id.value}`, {
  immediate: false
})
```

Reply:
The hell, GitHub says the comment is outdated.